package camera_utils.msgs;

import "vector3d.proto";

message CameraUtilsRequest
{
    /** Type of request */
    enum Type{
        /** Acquire frame and save it to disk */
        CAPTURE         = 1;
        /** Change output directory */
        DIR             = 2;
        /** From 3d to 2d camera point */
        CAMERA_POINT         = 3;
    }

    /** Type of request */
    required Type   type        = 1;
    /** Name of the file corresponding to the acquired frame */
    optional string file_name   = 2;
    /** Path to the output folder */
    optional string output_dir  = 3;
    /** 3D Point */
    optional gazebo.msgs.Vector3d point = 4;
    /* @brief Model name */
    optional string name = 5;
}
